torchaudio/datasets/datasets.py

```diff
@@ -0,0 +1,419 @@
+        yield {
+            "id": fileid,
+            "content": content,
+            "waveform": waveform,
+            "sample_rate": sample_rate,
+        }
+
+
+def VCTK(root):
+    """
+    Cache a pipeline loading VCTK.
+    """
+
+    url = [
+        (
+            "http://homepages.inf.ed.ac.uk/jyamagis/release/VCTK-Corpus.tar.gz",
+            "VCTK-Corpus/",
+        )
+    ]
+
+    path = download(url, root_path=root)
+    path = extract(path)
+    path = walk(path, extension=".wav")
+    path = shuffle(path)
+    data = load_vctk(path)
+
+    return Cache(data, "tmp/")
+
+
+def load_librispeech(fileids):
+    """
+    Load data corresponding to each LIBRISPEECH fileids.
+
+    Input: path, file name identifying a row of data
+    Output: id, waveform, sample_rate, translation
+    """
+
+    text_extension = ".trans.txt"
+    audio_extension = ".flac"
+    for data_path, fileid in fileids:
+        fileid = os.path.basename(fileid).split(".")[0]
+        folder1, folder2, file = fileid.split("-")
+        file_text = folder1 + "-" + folder2 + text_extension
+        file_text = os.path.join(data_path, folder1, folder2, file_text)
+        file_audio = folder1 + "-" + folder2 + "-" + file + audio_extension
+        file_audio = os.path.join(data_path, folder1, folder2, file_audio)
+        waveform, sample_rate = torchaudio.load(file_audio)
+
+        found = False
+        for line in open(file_text):
+            fileid_text, content = line.strip().split(" ", 1)
+            if fileid == fileid_text:
+                found = True
+                break
+        if not found:
+            from warnings import warn
+
+            warn("Translation not found for {}.".format(fileid))
+            continue
+
+        yield {
+            "id": fileid,
+            "content": content,
+            "waveform": waveform,
+            "sample_rate": sample_rate,
+        }
+
+
+def LIBRISPEECH(root, selection="dev-clean"):
+    """
+    Cache a pipeline loading LIBRISPEECH.
+    """
+
+    # http://www.openslr.org/resources/12/dev-clean.tar.gz
+    # http://www.openslr.org/resources/12/test-clean.tar.gz
+    # http://www.openslr.org/resources/12/test-other.tar.gz
+    # http://www.openslr.org/resources/12/train-clean-100.tar.gz
+    # http://www.openslr.org/resources/12/train-clean-360.tar.gz
+    # http://www.openslr.org/resources/12/train-other-500.tar.gz
+
+    selections = [
+        "dev-clean",
+        "test-clean",
+        "test-other",
+        "train-clean-100",
+        "train-clean-360",
+        "train-other-500",
+    ]
+
+    base = "http://www.openslr.org/resources/12/"
+    url = [
+        (
+            os.path.join(base, selection + ".tar.gz"),
+            os.path.join("LibriSpeech", selection),
+        )
+    ]
+
+    path = download(url, root_path=root)
+    path = extract(path)
+    path = walk(path, extension=".flac")
+    path = shuffle(path)
+    data = load_librispeech(path)
+
+    return Cache(data, "tmp/")
```

Comment:
I'd suggest we don't Cache this by default to enable streaming applications if people want. If it's past of a class we could make random access (caching) a flag.

Reply:
agree. this file wasn't supposed to be in the PR :)